packages/back-office-subscribers/nsip-representation/index.js

```diff
@@ -9,71 +10,49 @@ module.exports = async (context, message) => {
-			return;
-		}
-		let representation = {
-			representationId: message.representationId,
-			caseReference: message.caseRef,
-			caseId: message.caseId,
-			referenceId: message.referenceId,
-			status: message.status,
-			dateReceived: message.dateReceived,
-			representationComment: message.redacted
-				? message.redactedRepresentation
-				: message.originalRepresentation,
-			representationFrom: message.representationFrom,
-			representationType: message.representationType,
-			registerFor: message.registerFor,
-			attachmentIds: message.attachmentIds?.join(','),
-			modifiedAt: new Date()
-		};
-
-		if (message.representedId) {
-			representation.represented = {
-				connectOrCreate: {
-					where: {
-						serviceUserId: message.representedId
-					},
-					create: {
-						serviceUserId: message.representedId
-					}
-				}
-			};
-		}
-		if (message.representativeId) {
-			representation.representative = {
-				connectOrCreate: {
-					where: {
-						serviceUserId: message.representativeId
-					},
-					create: {
-						serviceUserId: message.representativeId
-					}
-				}
-			};
-		}
+	// Only create the service users if it doesn't already exist
+	// We create this first so that the foreign key constraint doesn't fail
+	if (message.representedId) {
+		await prismaClient.$executeRawUnsafe(serviceUserQuery, message.representedId);
+		context.log(`created represented with serviceUserId ${message.representedId}`);
+	}
+	if (message.representativeId) {
+		await prismaClient.$executeRawUnsafe(serviceUserQuery, message.representativeId);
+		context.log(`created representative with serviceUserId ${message.representativeId}`);
+	}
 
-		await tx.representation.upsert({
-			where: {
-				representationId
-			},
-			create: representation,
-			update: representation
-		});
+	let representation = {
+		representationId: message.representationId,
+		caseReference: message.caseRef,
+		caseId: message.caseId,
+		referenceId: message.referenceId,
+		status: message.status,
+		dateReceived: message.dateReceived,
+		representationComment: message.redacted
+			? message.redactedRepresentation
+			: message.originalRepresentation,
+		representationFrom: message.representationFrom,
+		representationType: message.representationType,
+		registerFor: message.registerFor,
+		attachmentIds: message.attachmentIds?.join(','),
+		representedId: message.representedId,
+		representativeId: message.representativeId,
+		modifiedAt: new Date()
+	};
 
-		context.log(`upserted representation with representationId: ${representationId}`);
-	});
+	const { statement, parameters } = buildMergeQuery(
+		'representation',
+		'representationId',
+		representation,
+		context.bindingData.enqueuedTimeUtc
+	);
+	await prismaClient.$executeRawUnsafe(statement, ...parameters);
+	context.log(`upserted representation with representationId ${representationId}`);
 };
+
+const serviceUserQuery = `
```

Comment:
This one is slightly different as we don't check for dates and it will only insert, the update doesn't actually do anything. If this is used in more places, we can definitely make `dateToCompare` optional in `buildMergeQuery` and make it more generic.